atoma-sui/src/handlers.rs

```diff
@@ -1,85 +1,492 @@
+///
+/// This function will return an error if:
+/// * The event data cannot be deserialized into a `NodeSubscribedToTaskEvent`.
+/// * The database operation to subscribe the node to the task fails.
+#[instrument(level = "trace", skip_all)]
+pub(crate) async fn handle_node_task_subscription_event(
+    value: Value,
+    db: &SqlitePool,
+) -> Result<()> {
+    trace!("Processing node subscription event");
+    let node_subscription_event: NodeSubscribedToTaskEvent = serde_json::from_value(value)?;
+    let node_small_id = node_subscription_event.node_small_id.inner as i64;
+    let task_small_id = node_subscription_event.task_small_id.inner as i64;
+    let price_per_compute_unit = node_subscription_event.price_per_compute_unit as i64;
+    let max_num_compute_units = node_subscription_event.max_num_compute_units as i64;
+    let state_manager = StateManager::new(db.clone());
+    state_manager
+        .subscribe_node_to_task(
+            node_small_id,
+            task_small_id,
+            price_per_compute_unit,
+            max_num_compute_units,
+        )
+        .await?;
+    Ok(())
+}
+
+/// Handles a node task subscription updated event.
+///
+/// This function processes a node task subscription updated event by parsing the event data,
+/// extracting the necessary information, and updating the node's subscription to the task in the database.
+///
+/// # Arguments
+///
+/// * `value` - A `serde_json::Value` containing the serialized node task subscription updated event data.
+/// * `db` - A reference to the `SqlitePool` for database operations.
+///
+/// # Returns
+///
+/// * `Result<()>` - Ok(()) if the event was processed successfully, or an error if something went wrong.
+///
+/// # Errors
+///
+/// This function will return an error if:
+/// * The event data cannot be deserialized into a `NodeSubscriptionUpdatedEvent`.
+/// * The database operation to update the node's subscription to the task fails.
+#[instrument(level = "trace", skip_all)]
+pub(crate) async fn handle_node_task_subscription_updated_event(
+    value: Value,
+    db: &SqlitePool,
+) -> Result<()> {
+    trace!("Processing node subscription updated event");
+    let node_subscription_event: NodeSubscriptionUpdatedEvent = serde_json::from_value(value)?;
+    let node_small_id = node_subscription_event.node_small_id.inner as i64;
+    let task_small_id = node_subscription_event.task_small_id.inner as i64;
+    let price_per_compute_unit = node_subscription_event.price_per_compute_unit as i64;
+    let max_num_compute_units = node_subscription_event.max_num_compute_units as i64;
+    let state_manager = StateManager::new(db.clone());
+    state_manager
+        .update_node_subscription(
+            node_small_id,
+            task_small_id,
+            price_per_compute_unit,
+            max_num_compute_units,
+        )
+        .await?;
+    Ok(())
+}
+
+#[instrument(level = "trace", skip_all)]
+pub(crate) async fn handle_node_task_unsubscription_event(
```

Comment:
this function doesnt have the nice comment/description as you have everywhere else

Reply:
Oh nice catch ! I was looking for these culprits !